server/aesdsocket.c

```diff
@@ -1,331 +1,580 @@
-/*
- * Author: Pranav Bharadwaj
- * Date: 09/30/2022
- * Formatted for space using codebeautify.org
- * References:
- 
- 1) Getting IPv4 address: https://stackoverflow.com/questions/1276294/getting-ipv4-address-from-a-sockaddr-structure
- 2) https://beej.us/guide/bgnet/html/#audience
- 3) https://www.tutorialspoint.com/c_standard_library/c_function_strerror.htm 
-*/
-
+/*Assignment 6 aesdsocket implementation*/
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <stdbool.h>
 #include <syslog.h>
+#include "queue.h"
+#include <fcntl.h>
 #include <errno.h>
 #include <unistd.h>
 #include <signal.h>
-#include <fcntl.h>
-#include <arpa/inet.h>
-#include <sys/types.h>
 #include <sys/socket.h>
+#include <sys/types.h>
 #include <netdb.h>
-#include <string.h>
+#include <arpa/inet.h>
+#include <pthread.h>
+#include <time.h>
+
+#define     QUEUE_LENGTH    (10)
+#define     LOGFILE_PATH  ("/var/tmp/aesdsocketdata")
+#define     BUFFER_SIZE   (1024)
+
+struct thread_t
+{
+    pthread_t thread_id;
+    int client_socket_t;
+    int server_socket_t;
+    int logfile_fd;
+    pthread_mutex_t *mutex;
+    bool complete;
+    int *file_size;
+    struct sockaddr_storage client_address;
+
+    //singly linked list node 
+    SLIST_ENTRY(thread_t) node;
+};
+
+//Global variables
+bool graceful_exit_handle = false;
+int file_fd = 0;
+int file_size = 0;
+
+//Mutex initialization
+pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
@prvbharadwaj1731 what is this mutex locking?  Suggest using a more descriptive name here.